src/sql/src/plan/query.rs

```diff
@@ -2094,6 +2107,119 @@ fn plan_table_factor(
     )
 }
 
+// The following SQL is equivalent:
```

Comment:
I really like the expression of the AST rewrite in very concrete terms. What would you think about including an expression of the approach in prose, as well? Could be brief but will certainly help folks years from now grok how this works a little more quickly.